src/test/java/org/folio/rest/impl/CleanUpJobTest.java

```diff
@@ -0,0 +1,205 @@
+        try (Connection connection = SingleConnectionProvider.getConnection(vertx, OAI_TEST_TENANT)) {
+          connection.prepareStatement("create schema oaitest_mod_oai_pmh")
+            .execute();
+        } catch (Exception ex) {
+          testContext.failNow(ex);
+        }
+        LiquibaseUtil.initializeSchemaForTenant(vertx, OAI_TEST_TENANT);
+        new OkapiMockServer(vertx, mockPort).start(testContext);
+        testContext.completeNow();
+      } catch (Exception e) {
+        testContext.failNow(e);
+      }
+    }));
+  }
+
+  @BeforeEach
+  private void initSampleData(VertxTestContext testContext) {
+    OffsetDateTime firstNotExpiredDate = new Date().toInstant()
+      .atOffset(ZoneOffset.of(ZoneId.systemDefault()
+        .getId()));
+    OffsetDateTime secondNotExpiredDate = new Date().toInstant()
+      .atOffset(ZoneOffset.of(ZoneId.systemDefault()
+        .getId()));
+    OffsetDateTime firstExpiredDate = new Date().toInstant()
+      .atOffset(ZoneOffset.of(ZoneId.systemDefault()
+        .getId()))
+      .minusSeconds(INSTANCES_EXPIRATION_TIME_IN_SECONDS);
+    OffsetDateTime secondExpiredDate = new Date().toInstant()
+      .atOffset(ZoneOffset.of(ZoneId.systemDefault()
+        .getId()))
+      .minusSeconds(INSTANCES_EXPIRATION_TIME_IN_SECONDS);
+
+    RequestMetadataLb firstExpiredRequestMetadata = new RequestMetadataLb().setId(UUID.fromString(EXPIRED_REQUEST_ID_1))
+      .setRequestId(EXPIRED_REQUEST_ID_1)
+      .setLastUpdatedDate(firstExpiredDate);
+    RequestMetadataLb secondExpiredRequestMetadata = new RequestMetadataLb().setId(UUID.fromString(EXPIRED_REQUEST_ID_2))
+      .setRequestId(EXPIRED_REQUEST_ID_2)
+      .setLastUpdatedDate(secondExpiredDate);
+    RequestMetadataLb firstNotExpiredRequestMetadata = new RequestMetadataLb().setId(UUID.fromString(NOT_EXPIRED_REQUEST_ID_3))
+      .setRequestId(NOT_EXPIRED_REQUEST_ID_3)
+      .setLastUpdatedDate(firstNotExpiredDate);
+    RequestMetadataLb secondNotExpiredRequestMetadata = new RequestMetadataLb().setId(UUID.fromString(NOT_EXPIRED_REQUEST_ID_3))
+      .setRequestId(NOT_EXPIRED_REQUEST_ID_3)
+      .setLastUpdatedDate(secondNotExpiredDate);
+
+    List<Future> futures = new ArrayList<>();
+    List
+      .of(firstExpiredRequestMetadata, secondExpiredRequestMetadata, firstNotExpiredRequestMetadata,
+          secondNotExpiredRequestMetadata)
+      .forEach(elem ->
+        futures.add(saveRequestMetadata(elem)));
+
+    CompositeFuture.all(futures)
+      .onSuccess(reply -> testContext.completeNow())
+      .onFailure(testContext::failNow);
+  }
+
+  @Test
+  void shouldReturn204AndClearExpiredInstances(VertxTestContext testContext) {
+    RequestSpecification request = createBaseRequest(CLEAN_UP_INSTANCES_PATH, null);
+    request.when()
+      .post()
+      .then()
+      .statusCode(204);
+  }
+
+  //will add batch save
+  private Future<RequestMetadataLb> saveRequestMetadata(RequestMetadataLb requestMetadata) {
```

Comment:
Need to be moved to the dao layer and be reused at MarcWithHoldingsRequestHelper and test classes as well.